File: api/src/main/java/be/gilmotech/nestspend/dto/futureevent/ProjectionDataPoint.java
package be.gilmotech.nestspend.dto.futureevent;

import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDate;

/**
 * Data point for budget projection.
 */
@Schema(description = "Single data point in budget projection")
public record ProjectionDataPoint(
        @Schema(description = "Date of this data point", example = "2025-01-31")
        LocalDate date,

        @Schema(description = "Projected total income in cents for this period", example = "500000")
        Long projectedIncomeCents,

        @Schema(description = "Projected total expenses in cents for this period", example = "350000")
        Long projectedExpenseCents,

        @Schema(description = "Projected balance in cents at this date", example = "150000")
        Long projectedBalanceCents
) {
}

File: web/nestspend-web/src/app/core/api/models/projection-data-point.ts
/* eslint-disable */

/**
 * Single data point in budget projection
 */
export interface ProjectionDataPoint {
  /**
   * Date of this data point
   */
  date?: string;

  /**
   * Projected total income in cents for this period
   */
  projectedIncomeCents?: number;

  /**
   * Projected total expenses in cents for this period
   */
  projectedExpenseCents?: number;

  /**
   * Projected balance in cents at this date
   */
  projectedBalanceCents?: number;
}
